### ISSUE TITLE
[Question] How can I specify schema name?

### ISSUE THREAD

**Issue by mdrgeniebelt** [ch-robinson/dotnet-avro#128]
Hi,

I'm trying to use your library for Avro schema registry but I encountered few issues.

This is my code for producer builder:

```c#
var contractName = PascalToKebabCase(typeof(TContract).Name);
var kafkaProducer = new ProducerBuilder<TKey, MessageEnvelope<TContract>>(_producerConfig)
    .SetAvroValueSerializer(_schemaRegistryClient, AutomaticRegistrationBehavior.Always,
        context => context.Component == MessageComponentType.Key ? $"{contractName}-key" : contractName)
    .Build();
```
As you can see our messages are wrapped in `MessageEnvelope<T>` which contains additional fields with metadata (trace, user information etc). 

Now when I try to publish a message it throws exception:

```
Confluent.Kafka.ProduceException`2[System.String,LetsBuild.Kafka.Abstractions.Contracts.MessageEnvelope`1[KafkaTestApp.AccountEvent]]: Local: Value serialization error
 ---> Chr.Avro.Abstract.InvalidNameException: "LetsBuild.Kafka.Abstractions.Contracts.MessageEnvelope`1" is not a valid Avro name.
   at Chr.Avro.Abstract.NamedSchema.set_FullName(String value)
   at Chr.Avro.Abstract.NamedSchema..ctor(String name)
   at Chr.Avro.Abstract.RecordSchema..ctor(String name, ICollection`1 fields)
   at Chr.Avro.Abstract.RecordSchemaBuilderCase.BuildSchema(TypeResolution resolution, ConcurrentDictionary`2 cache)
   at Chr.Avro.Abstract.SchemaBuilder.BuildSchema(Type type, ConcurrentDictionary`2 cache)
   at Chr.Avro.Abstract.SchemaBuilder.BuildSchema[T](ConcurrentDictionary`2 cache)
   at Chr.Avro.Confluent.AsyncSchemaRegistrySerializer`1.<SerializeAsync>b__26_0(String subject)
   at Chr.Avro.Confluent.AsyncSchemaRegistrySerializer`1.SerializeAsync(T data, SerializationContext context)
   at Confluent.Kafka.Producer`2.ProduceAsync(TopicPartition topicPartition, Message`2 message, CancellationToken cancellationToken)
```

Is there a way to specify the name? Maybe I'm misunderstanding what `subjectNameBuilder` function is responsible of setting?

We would like to control how the schemas are called and also the name property in the avro schema.

Thanks in advance!

**Comment by dstelljes** [comment]
If you want to override the names of records or their fields, you can use the `[DataContract]` and `[DataMember]` attributes. By default, the schema builder will use the name/namespace properties of those attributes.

The exception you’re seeing is because of the`` `1`` from the generic `MessageEnvelope` type name. The schema builder should handle those generic names more gracefully, so I’ll leave this open as a bug.

**Comment by dstelljes** [comment]
The fix for the generic name issue (#187) is released with 8.0.0-pre.1.
